beamer/check/commands.py

```diff
@@ -264,3 +264,166 @@ def initiate_l1_invalidations(
+            )
+            continue
+
+        # First check the finalization timestamp.
+        if (
+            invalidation.finalization_timestamp is None
+            or datetime.utcnow() <= invalidation.finalization_timestamp
+        ):
+            log.error(
+                "Invalidation not yet finalized, skipping",
+                idx=idx,
+                proof_source=pair[0],
+                proof_target=pair[1],
+            )
+            errors += 1
+            continue
+
+        deployment = beamer.artifacts.load(artifacts_dir, invalidation.proof_target)
+        url = rpc_info[invalidation.proof_target]
+        w3 = make_web3(url, account)
+        assert w3.eth.chain_id == invalidation.proof_target
+        log.info("Connected to RPC", url=url)
+
+        # Check the request manager directly to see if it has been already verifed.
+        request_manager = obtain_contract(w3, abi_manager, deployment, "RequestManager")
+        if request_manager.functions.isInvalidFill(
+            invalidation.request_id, invalidation.fill_id
+        ).call():
+            log.info(
+                "Invalidation verified successfully",
+                idx=idx,
+                proof_source=pair[0],
+                proof_target=pair[1],
+            )
+            verified.add(pair)
+            continue
+
+        log.debug("Starting relayer", txhash=invalidation.txhash)
+        try:
+            run_relayer_for_tx(
+                l1_rpc=rpc_info[deployment.base.chain_id],
+                l2_relay_from_rpc_url=rpc_info[invalidation.proof_source],
+                l2_relay_to_rpc_url=rpc_info[invalidation.proof_target],
+                account=account,
+                tx_hash=invalidation.txhash,
+            )
+        except RelayerError as exc:
+            log.error(
+                "Relayer failed", idx=idx, exc=exc, proof_source=pair[0], proof_target=pair[1]
+            )
+            errors += 1
+        else:
+            log.info("Relayer succeeded", thxash=invalidation.txhash)
+
+            # Check the request manager, the invalidation should have arrived.
+            if request_manager.functions.isInvalidFill(
+                invalidation.request_id, invalidation.fill_id
+            ).call():
+                log.info(
+                    "Invalidation verified successfully",
+                    idx=idx,
+                    proof_source=pair[0],
+                    proof_target=pair[1],
+                )
+                verified.add(pair)
+            else:
+                log.error(
+                    "Invalidation failed",
+                    idx=idx,
+                    proof_source=pair[0],
+                    proof_target=pair[1],
+                )
+                errors += 1
+
+    if errors:
+        log.error(
+            "Error(s) occurred during verification",
+            errors=errors,
+            verified=len(verified),
+            chain_pairs=chain_pairs,
```

Comment:
is this really true? I think it would confuse me, because chain_pairs contains all chain_pairs which have been touched regardless of error or success

Reply:
Yeah, that's confusing, I wanted to give info like "verified M out of N chain pairs in total", something like that....
Maybe something like this?
```python
            total_chain_pairs=len(chain_pairs)
```

Reply:
That would be better, But I probably would also like to see the failing chain pairs, for further debugging

Reply:
Addressed this in the new version, by showing all unverified chain pairs. The logs will also contain specific errors that occurred.